spec/concepts/api/v1/users/registrations/contract/create_spec.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+RSpec.describe Api::V1::Users::Registrations::Contract::Create do
+  subject(:contract) do
+    described_class
+  end
+
+  let(:result) { contract.call(params) }
+
+  describe 'User with invalid credentials' do
+    context 'when missed required fields' do
+      let(:params) { {} }
+
+      it 'validates presence of the required keys' do
+        expect(result.errors.to_h[:email]).to include('is missing')
+        expect(result.errors.to_h[:password]).to include('is missing')
+        expect(result.errors.to_h[:password_confirmation]).to include('is missing')
+      end
+    end
+
+    context 'when required fields are blank' do
+      let(:params) { { email: '', password: '', password_confirmation: '' } }
+
+      it 'validates presence of the required fields' do
+        expect(result.errors.to_h[:email]).to include('must be filled')
+        expect(result.errors.to_h[:password]).to include('must be filled')
+        expect(result.errors.to_h[:password_confirmation]).to include('must be filled')
+      end
+    end
+
+    context 'when required fields has unexpected types' do
+      let(:password) { 12_345_678 }
+      let(:params) { { email: [random_email], password: password, password_confirmation: password } }
+
+      it 'validates that required fields have expected types' do
+        expect(result.errors.to_h[:email]).to include('must be a string')
+        expect(result.errors.to_h[:password]).to include('must be a string')
+        expect(result.errors.to_h[:password_confirmation]).to include('must be a string')
+      end
+    end
+  end
+
+  describe 'User can not be saved' do
```

Comment:
What??? It's not a responsibility of contract to persist user.

Reply:
Ok I'll try to test unit in isolation